expression/builtin_time_test.go

```diff
@@ -1610,37 +1610,35 @@ func (s *testEvaluatorSuite) TestMakeTime(c *C) {
 		{[]interface{}{12, 15, -30}, nil},
 
 		{[]interface{}{12, 15, "30.10"}, "12:15:30.100000"},
-		{[]interface{}{12, 15, "30.00"}, "12:15:30.000000"},
-		{[]interface{}{12, 15, 30.0000001}, "12:15:30.000000"},
+		{[]interface{}{12, 15, "30.20"}, "12:15:30.200000"},
+		{[]interface{}{12, 15, 30.3000001}, "12:15:30.300000"},
 		{[]interface{}{12, 15, 30.0000005}, "12:15:30.000001"},
-		{[]interface{}{"12", "15", 30.1}, "12:15:30.1"},
+		{[]interface{}{"12", "15", 30.1}, "12:15:30.100000"},
 
 		{[]interface{}{0, 58.4, 0}, "00:58:00"},
 		{[]interface{}{0, "58.4", 0}, "00:58:00"},
 		{[]interface{}{0, 58.5, 1}, "00:59:01"},
 		{[]interface{}{0, "58.5", 1}, "00:58:01"},
 		{[]interface{}{0, 59.5, 1}, nil},
 		{[]interface{}{0, "59.5", 1}, "00:59:01"},
-		{[]interface{}{0, 1, 59.1}, "00:01:59.1"},
+		{[]interface{}{0, 1, 59.1}, "00:01:59.100000"},
 		{[]interface{}{0, 1, "59.1"}, "00:01:59.100000"},
-		{[]interface{}{0, 1, 59.5}, "00:01:59.5"},
+		{[]interface{}{0, 1, 59.5}, "00:01:59.500000"},
 		{[]interface{}{0, 1, "59.5"}, "00:01:59.500000"},
 		{[]interface{}{23.5, 1, 10}, "24:01:10"},
 		{[]interface{}{"23.5", 1, 10}, "23:01:10"},
 
 		{[]interface{}{0, 0, 0}, "00:00:00"},
-		{[]interface{}{"", "", ""}, "00:00:00.000000"},
-		{[]interface{}{"h", "m", "s"}, "00:00:00.000000"},
 
-		{[]interface{}{837, 59, 59.1}, "837:59:59.1"},
-		{[]interface{}{838, 59, 59.1}, "838:59:59.0"},
-		{[]interface{}{-838, 59, 59.1}, "-838:59:59.0"},
+		{[]interface{}{837, 59, 59.1}, "837:59:59.100000"},
```

Comment:
in mysql:
```sql
MySQL > select maketime(837, 59, 59.1);
-------------------------+
| maketime(837, 59, 59.1) |
+-------------------------+
| 837:59:59.1             |
+-------------------------+
1 row in set (0.00 sec)

```

Reply:
Yep. You can see my integration tests.There's a similar case and the result is correct. Decimals goes wrong here because 'datumsToConstants' don't correctly set constants's decimal(all are set -1, which is not as expected). So here the tests equal to creating a table with a double column, and use it as the third parameter in MAKETIME

Reply:
@zz-json